pkg/model/resource/options.go

```diff
@@ -228,9 +230,15 @@ func (opts *Options) safeImport(unsafe string) string {
 }
 
 // NewResource creates a new resource from the options
-func (opts *Options) NewResource(c *config.Config, doResource bool) *Resource {
+func (opts *Options) NewResource(c *config.Config, doResource, doController bool) *Resource {
 	res := opts.newResource()
 
+	// the following check was only add to ensure that the behaviour would not
+	// be changed for v2 projects. This if can be removed when v2 is no longer supported
+	if !c.IsV2() {
+		res.Controller = doController
+	}
```

Comment:
Setting any resource fields should be encapsulated in related methods, ex
```suggestion
	res := opts.newResource(doController)
```

Reply:
the `opts.newResource` has not the config. It returns a new resource from opts and not with all data.